app/controllers/DeclarationController.scala

```diff
@@ -16,35 +16,47 @@
 
 package controllers
 
-import javax.inject.Inject
-import play.api.i18n.{I18nSupport, MessagesApi}
-import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import config.FrontendAppConfig
 import connectors.DataCacheConnector
 import controllers.actions._
-import config.FrontendAppConfig
-import models.Mode
-import pages.ConfirmationPage
+import javax.inject.Inject
+import mapper.CaseRequestMapper
+import models.Confirmation.format
+import models._
 import navigation.Navigator
+import pages.ConfirmationPage
+import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.mvc.{Action, AnyContent}
+import service.CasesService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import views.html.declaration
 
-import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
 
 class DeclarationController @Inject()(
-                                        appConfig: FrontendAppConfig,
-                                        override val messagesApi: MessagesApi,
-                                        dataCacheConnector: DataCacheConnector,
-                                        navigator: Navigator,
-                                        identify: IdentifierAction,
-                                        getData: DataRetrievalAction
-                                      ) extends FrontendController with I18nSupport {
+                                       appConfig: FrontendAppConfig,
+                                       override val messagesApi: MessagesApi,
+                                       dataCacheConnector: DataCacheConnector,
+                                       navigator: Navigator,
+                                       identify: IdentifierAction,
+                                       getData: DataRetrievalAction,
+                                       service: CasesService
+                                     ) extends FrontendController with I18nSupport {
 
   def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData) { implicit request =>
     Ok(declaration(appConfig, mode))
   }
 
   def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData).async { implicit request =>
-    Future.successful(Redirect(navigator.nextPage(ConfirmationPage, mode)(request.userAnswers.get)))
+    val answers: UserAnswers = request.userAnswers.get
+
+    service
+      .createCase(new CaseRequestMapper().map(answers))
```

Comment:
We should not create a new instance of `CaseRequestMapper`
You rather have an object (rather than a class) or a singleton instance you inject it here

Reply:
Woops forgot about this, I meant to make it a Singleton class

Reply:
Sorted